src/tools/cephfs/first-damage.py

```diff
@@ -59,7 +59,7 @@
 CONF = os.environ.get('CEPH_CONF')
 REPAIR_NOSNAP = None
 
-CEPH_NOSNAP = 0xfffffffe # int32 -2
+CEPH_NOSNAP = 0xfffffffffffffffe # int64 -2
```

Comment:
I'm not sure why I used int32 here when clearly in the code it is indeed an int64. Can you please dumps of the inode_t with ceph-dencoder before and after this patch to demonstrate this is indeed correct.

Reply:
The inode dump is (first few interesting lines)

```
00000000  e1 0b 00 00 00 01 00 00  69 02 01 09 06 00 00 00  |........i.......|
00000010  00 00 00 06 04 ff 05 00  00 11 06 b5 01 00 00 26  |...............&|
00000020  30 bb 03 00 01 00 00 00  00 00 00 44 c5 4e 66 7f  |0..........D.Nf.|
00000030  2e 45 08 f8 81 00 00 cb  49 26 00 81 86 1e 00 01  |.E......I&......|
00000040  00 00 00 00 00 00 00 00  00 00 00 00 02 02 18 00  |................|
00000050  00 00 00 00 40 00 01 00  00 00 00 00 40 00 0f 00  |....@.......@...|
00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000070  00 00 01 00 00 00 ff ff  ff ff ff ff ff ff 00 00  |................|
```

(couldn't get ceph-dencode to dump the inode)

The first field seems corrupted: `10000000be1' which is seen in the MDS log.

And @kotreshhr then ran this through the python shell

```
>>> val=b'\xe1\x0b\x00\x00\x00\x01\x00\x00\x69\x02\x01\x09\x06\x00\x00\x00'
>>> (first,) = struct.unpack('<I', val[:4])
>>> print(first)
3041
>>> (first,) = struct.unpack('<Q', val[:8])
>>> print(first)
1099511630817
>>> print(hex(first))
0x10000000be1
>>>
```